pallets/pallet-credential/src/lib.rs

```diff
@@ -75,59 +74,83 @@ decl_event!(
+    CredInvalidSubject,
+    CredTooManyProps,
     }
 }
 
 decl_module! {
     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
-        type Error = Error<T>;
-        fn deposit_event() = default;
-
-        #[weight = 10]
-        pub fn register_credential(origin, id: CredId, owner: T::AccountId, hash: T::Hash, 
-            props: Option<Vec<CredProperty>>) -> dispatch::DispatchResult {
-
-            T::CreateRoleOrigin::ensure_origin(origin.clone())?;
-            let who = ensure_signed(origin)?;
-            
-            // Validate Cred ID
-            Self::validate_cred_id(&id)?;
-
-            // Validate credential props
-            Self::validate_cred_props(&props)?;
-
-            // Check credential doesn't exist yet (1 DB read)
-            Self::validate_new_credential(&id)?;
-
-            // Create a credential instance
-            let credential = Self::new_credential()
-                .identified_by(id.clone())
-                .owned_by(owner.clone())
-                .credential_hash(hash.clone())
-                .registered_on(<timestamp::Module<T>>::now())
-                .with_props(props)
-                .build();
-
-            // Add Cred & ownerOf (4 DB writes)
-            <Credentials<T>>::insert(&id, credential);
-            <CredsOfOrganization<T>>::append(&owner, &id);
-            <IssuerOfCred<T>>::insert(&id, &owner);
-            <CredByHash<T>>::append(&hash, &id);
-            
-            Self::deposit_event(RawEvent::CredentialRegistered(who, id, hash));
-
-            Ok(())
-        }
+    type Error = Error<T>;
+    fn deposit_event() = default;
+
+    #[weight = 10]
+    pub fn register_credential(origin, id: CredId, owner: T::AccountId, hash: T::Hash,
+        props: Option<Vec<CredProperty>>) -> dispatch::DispatchResult {
+
+        T::CreateRoleOrigin::ensure_origin(origin.clone())?;
+        let who = ensure_signed(origin)?;
+
+        // Validate Cred ID
+        Self::validate_cred_id(&id)?;
+
+        // Validate credential props
+        Self::validate_cred_props(&props)?;
+
+        // Check credential doesn't exist yet (1 DB read)
+        Self::validate_new_credential(&id)?;
+
+        // Create a credential instance
+        let credential = Self::new_credential()
+        .identified_by(id.clone())
+        .owned_by(owner.clone())
+        .credential_hash(hash.clone())
+        .registered_on(<timestamp::Module<T>>::now())
+        .with_props(props)
+        .build();
+
+        // Add Cred & ownerOf (4 DB writes)
+        <Credentials<T>>::insert(&id, credential);
+        <CredsOfOrganization<T>>::append(&owner, &id);
+        <IssuerOfCred<T>>::insert(&id, &owner);
+        <CredByHash<T>>::append(&hash, &id);
+
+        Self::deposit_event(RawEvent::CredentialRegistered(who, id, hash));
+
+        Ok(())
+    }
+
+    #[weight = 100]
+    pub fn revoke_credential(origin, id: CredId, owner: T::AccountId, hash: T::Hash) -> dispatch::DispatchResult {
```

Comment:
register and revoke got the same signature. how do we differentiate the response?

Reply:
Was referencing https://github.com/substrate-developer-hub/substrate-verifiable-credentials/blob/master/assets/5.2-finished-code.rs#L42 while doing it.

Will add tests and see how it works.